### ISSUE TITLE
How to define custom loss in loss.py

### ISSUE THREAD

**Issue by NCTU-VRDL** [open-mmlab/mmdetection#2684]
Hi! 

I need to implement custom loss to train the detectors. Since there is some non-differentiable operation in my loss function, I need to use [torch.extend_autograd](https://pytorch.org/docs/stable/notes/extending.html#extending-torch-autograd) to implement it.

By the document, it inherited from nn.Function, but all losses mmdetection are implemented by nn.Module and did not implement the backward function. I am wondering how to define my custom loss with self-defined backward correctly. Thanks!

**Comment by ZwwWayne** [comment]
See the implementation of [focal loss](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/losses/focal_loss.py)
